lib/dma.c

```diff
@@ -207,11 +218,6 @@ dma_controller_remove_regions(dma_controller_t *dma)
 void
 dma_controller_destroy(dma_controller_t *dma)
 {
-    if (dma == NULL) {
-        return;
-    }
-
-    dma_controller_remove_regions(dma);
```

Comment:
Sorry I think previous comment was not clear. I mean to say that why call to dma_controller_remove_regions() is been removed.